sdk/test-utils/recorder/README.md

```diff
@@ -1,9 +1,450 @@
+  request in a plain text file in the folder `recordings` at the root of your
+  repository (which for our example case is the root of the
+  `@azure/keyvault/keyvault-keys` repository). It must be used inside of Mocha's
+  tests, since it uses [nise](https://www.npmjs.com/package/nise) under the hood.
+  It also returns an object with a method `stop()`, which will allow you to control when
+  you want the recorder to stop re-routing your http requests.
+- `env`, which exposes the environment variable's object from either NodeJS or
+  the browser (useful on isomorphic tests).
+- `delay`, which is an asynchronous function that will resolve once the given milliseconds have elapsed,
+  but only while the `TEST_MODE` is not `playback`, since you want to make sure you can run the playback tests
+  as fast as possible.
+- `isRecordMode`, which is a shorthand for checking if the environment variable
+  `TEST_MODE` is set to `record`.
+- `isPlaybackMode`, which is a shorthand for checking if the environment
+  variable `TEST_MODE` is set to `playback`.
+- `setReplaceableVariables`, which will allow you to hide sensitive content
+  from the environment variables (more on that later).
+- `setReplacements`, which will allow you to hide sensitive content by doing
+  pattern matching in the output files.
+- `skipQueryParams`, since query parameters may contain sensitive information,
+  the array provided to method will signal what query parameters to remove from
+  the recordings.
+
+### Configuring your project
+
+Having the common recorder as a dependency means that you'll be able to start
+recording tests right away by using the exported method `record`. We'll get
+into the details further down this document. This function will do recordings,
+or will play back previous recordings, depending on an environment variable:
+`TEST_MODE`. If the environment variable `TEST_MODE` is empty, `record` (and most
+of the functions provided by test-utils-recorder) won't be doing anything. You'll need
+to set this environment variable to `record` to start recording, and then to
+`playback` to play the recordings back at your code.
+
+#### package.json scripts
+
+You can write scripts in your `package.json` to
+make it easier to switch from record mode to playback mode, on a meaningful context, as follows:
+
+    "integration-test:node": "mocha myNodeTests.js",
+    "unit-test:node": "TEST_MODE=playback npm run integration-test:node",
+    "test:node:record": "TEST_MODE=record npm run integration-test:node"
+
+
+Once your tests run, new files will be created in the `recordings/*` folder.
+These files will have names that are relative to the tests that you have.
+There might be cases in which the recordins get outdated with the test files, so you might also want to
+add a way to clear the recordings on your `package.json`, like the following one:
+
+    "clear-recordings": "rm -fr recordings",
+
+#### Karma configuration
```

Comment:
This might be enough for the basic setup.
To get the console.logs properly, more setup is required. If you add them here it'll be too verbose.
Maybe add a [this](https://github.com/Azure/azure-sdk-for-js/blame/master/sdk/test-utils/recorder/GUIDELINES.md#L240) link at the end of this section mentioning the issue with console.logs in browsers

Reply:
I will add it!